### DISCUSSION TITLE
Comment: "npm install" in Docker on Raspberry Pi 3

### DISCUSSION THREAD

**Discussion by Tamburasca** [iptv-org/community#514] [Q&A] [answered]
Just as an aside comment:
"npm install" will run into an error when executed in Docker on a Raspberry Pi 3 as 
resource https://github.com/marudor/libxmljs2/releases/download/v0.32.0/node-v108-linux-arm-musl.tar.gz
is missing. For sure it's a 32 bit system. Too bad...

**Comment by freearhey** [comment]
Changed `libxmljs2` version to `0.33.0` in https://github.com/iptv-org/epg/commit/7db92ae2f70983f95db0b2272730540d78e70ce2 maybe this will help.

**Reply by Tamburasca** [reply]
Thanx, I'll let you know

**Comment by Tamburasca** [comment]
Sorry, here's the error message I got:
 postinstall
> npm run api:load


> api:load
> npx tsx scripts/commands/api/load.ts

Error: Cannot find module 'fs/promises'
Require stack:
- /iptv/epg/node_modules/path-scurry/dist/cjs/index.js
- /iptv/epg/node_modules/@freearhey/core/node_modules/glob/dist/cjs/src/glob.js
- /iptv/epg/node_modules/@freearhey/core/node_modules/glob/dist/cjs/src/index.js
- /iptv/epg/node_modules/@freearhey/core/dist/storage.js
- /iptv/epg/node_modules/@freearhey/core/dist/index.js
- /iptv/epg/scripts/commands/api/load.ts
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:815:15)
    at Function.l.default._resolveFilename (/iptv/epg/node_modules/tsx/dist/cjs/index.cjs:1:1777)
    at Function.Module._load (internal/modules/cjs/loader.js:667:27)
    at Module.require (internal/modules/cjs/loader.js:887:19)
    at require (internal/modules/cjs/helpers.js:74:18)
    at /iptv/epg/node_modules/path-scurry/dist/cjs/index.js:35:20
    at Object.<anonymous> (/iptv/epg/node_modules/path-scurry/dist/cjs/index.js:3:3)
    at Module._compile (internal/modules/cjs/loader.js:999:30)
    at Object.S (/iptv/epg/node_modules/tsx/dist/cjs/index.cjs:1:1250)
    at Module.load (internal/modules/cjs/loader.js:863:32)
npm ERR! code 1
npm ERR! path /iptv/epg
npm ERR! command failed
npm ERR! command sh -c npm run api:load

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2023-11-16T17_30_54_702Z-debug-0.log
The command '/bin/sh -c npm install' returned a non-zero code: 1

Anyways, I appreciated your assistance....

**Comment by Tamburasca** [comment, marked as answer]
Finally I got docker to run on a Rasp PI (3 is not recommended)
This code may help (comprises two Web APIs)
++++

FROM ubuntu:22.04

WORKDIR /iptv/iptv/
COPY src/*.py src/
COPY requirements.txt .

RUN apt-get update \
    && apt-get upgrade -y
RUN apt-get install git -y
RUN apt-get install curl -y
RUN curl -fsSL https://deb.nodesource.com/setup_current.x | bash -
RUN apt-get install nodejs -y \
    && npm update --global npm
RUN apt-get install python3-pip -y \
    && pip install -r requirements.txt \
    && rm -f requirements.txt

WORKDIR /iptv/
COPY wrapper.sh .
RUN ["chmod", "+x", "wrapper.sh"]
RUN git clone --depth 1 -b master https://github.com/iptv-org/epg.git

WORKDIR /iptv/epg/
ARG MAX_OLD_SPACE_SIZE=512
ENV NODE_OPTIONS="--max_old_space_size=${MAX_OLD_SPACE_SIZE}"
RUN npm install

WORKDIR /iptv/
CMD ./wrapper.sh

+++ wrapper +++

cd /iptv/iptv/src/ || exit
python3 -u epg_extractor.py --port "$PORT" &

cd /iptv/epg/scripts/commands/epg/
npm run grab -- --site=tv.blue.ch --maxConnections=2
npm run serve &

wait
exit $?
